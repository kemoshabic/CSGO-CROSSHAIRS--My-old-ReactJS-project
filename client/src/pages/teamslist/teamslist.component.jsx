import React, { useEffect } from 'react';
import { connect } from 'react-redux';

import TeamsDirectoryContainer from '../../components/teams-directory/teams-directory.container'

import { fetchTeamsStart } from '../../redux/teams/teams.actions'

import './teamslist.styles.scss'

export const TeamsList = ({ fetchTeamsStart }) => {
  useEffect(() => {
    fetchTeamsStart();
  }, [fetchTeamsStart]);

  return (
    <div className="teams-page">
      <h3 className="teams-page__heading">Top 30 Teams (only top 8 teams provided with data)</h3>
      <TeamsDirectoryContainer />
    </div>
  )
}

const mapDispatchToProps = (dispatch) => ({
  fetchTeamsStart: () => dispatch(fetchTeamsStart()),
})

export default connect(null, mapDispatchToProps)(TeamsList);