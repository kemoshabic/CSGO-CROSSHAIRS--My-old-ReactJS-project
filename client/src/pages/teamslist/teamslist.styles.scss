.teams-page{
  padding: 10rem 15rem 10rem 15rem;
  @media only screen and (max-width: 103.12em ){
    padding: 6rem 3rem 5rem 3rem;
  }

  &__heading{
    font-size: 2.4rem;
    padding-bottom: 2rem;
    @media only screen and (max-width: 103.12em ){
      text-align: center;
      padding-bottom: 3rem;
    }
  }

}