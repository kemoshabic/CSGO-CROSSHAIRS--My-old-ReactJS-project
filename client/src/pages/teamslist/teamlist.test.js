import React from "react";
import { shallow } from "enzyme";
import { TeamsList } from "./teamslist.component";

describe("TeamsList Component", () => {
  let wrapper;
  let mockProps;
  // let useEffect;

  // const data = {
  //   teamName : {
  //   team: "text",
  //   imageUrl: "text",
  //   color: "text",
  //   players: ["a", "b", "c", "d", "e"],
  //   id: 1,
  //   teamsData: []
  //   }
  // }
  // const mockUseEffect = () => {
  //   useEffect.mockImplementationOnce(f => f());
  // };

  beforeEach(() => {
    // useEffect = jest.spyOn(React, "useEffect");

    mockProps = {
      // fetchTeamsStart: jest.fn().mockResolvedValue(data),
    };
    // mockUseEffect();
    wrapper = shallow(<TeamsList {...mockProps} />);
  });
  it("should render the TeamsList component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  // describe("on start", () => {
  //   it("call fetchTeamsStart fucntion", () => {
  //     expect(mockProps.fetchTeamsStart).toHaveBeenCalled();
  //   });
  // });
});
