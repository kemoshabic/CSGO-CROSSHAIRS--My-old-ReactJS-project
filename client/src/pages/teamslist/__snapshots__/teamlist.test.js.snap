// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamsList Component should render the TeamsList component 1`] = `ShallowWrapper {}`;
